### PR TITLE
[Master] Added some mock logic to SampleChangerMockup

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hi,

Just added some logic to simulate loading and unloading of samples a bit better.

Marcus

### CODE REVIEW

File: SampleChangerMockup.py [hunk 1]
```diff
@@ -23,23 +24,51 @@ def init(self):
...
+        self._triggerLoadedSampleChangedEvent(self.getLoadedSample())
+ 
     def getBasketList(self):
         basket_list = []
         for basket in self.components:
             if isinstance(basket, Basket):
                 basket_list.append(basket)
         return basket_list
 
+    def getLoadedSample(self):
```
Comment (IvarsKarpics) [inline SampleChangerMockup.py:63]: Hi @marcus-oscarsson 

getLoadedSample should return a Sample o Pin class and not a str. In the sample changer brick and dc_tree_widget it relies on this sample object.
I can revert to the previous version and add mockup load/unload logic.
Sorry for pointing this out after accepting the PR. I did not test it properly...

**Comment by IvarsKarpics** [comment]
See a solution #105

**Comment by marcus-oscarsson** [comment]
Thanks Ivars,

No problem, Matias merged #105 so I hope its as it should now !.

Marcus

**Comment by marcus-oscarsson** [comment]
Hi Ivars,

Ive merged #105 thanks for the fix :)

Marcus

**Comment by marcus-oscarsson** [comment]
Hi again,

Ill get back to you on the serialization work, I'm a bit busy at the moment so it will be a couple of days. But I'm sure we can do something :).

Marcus

**Pull request merged**